### PR TITLE
[Windows] Fix crash using animations and changing View sizes (negative values)

### LINKED ISSUE #7953: Animation crashes app when size is less than zero
### Description

This code works in Xamarin.Forms, but crashes in .NET MAUI. It is animating 2 `BoxView`s from 0 to some number with easing.

```
var inAnims = new Animation();
inAnims.Add(0, 1, new Animation(v => this.MilesProgressBar.WidthRequest = v, 0, 200, Easing.SpringIn));
inAnims.Add(0.25, 1, new Animation(v => this.SegmentsProgressBar.WidthRequest = v, 0, 100, Easing.SpringIn));

inAnims.Commit(this, "grow_bar", 16, 2000);
```

The crash error is:

```
06-11 16:38:28.916 14158 14158 E AndroidRuntime: FATAL EXCEPTION: main
06-11 16:38:28.916 14158 14158 E AndroidRuntime: Process: com.companyname.flyme, PID: 14158
06-11 16:38:28.916 14158 14158 E AndroidRuntime: android.runtime.JavaProxyThrowable: System.InvalidOperationException: Width cannot be less than zero.
06-11 16:38:28.916 14158 14158 E AndroidRuntime:    at Microsoft.Maui.Controls.VisualElement.ValidatePositive(Double value, String name)
06-11 16:38:28.916 14158 14158 E AndroidRuntime:    at Microsoft.Maui.Controls.VisualElement.Microsoft.Maui.IView.get_Width()
06-11 16:38:28.916 14158 14158 E AndroidRuntime:    at Microsoft.Maui.ViewHandlerExtensions.GetDesiredSizeFromHandler(IViewHandler viewHandler, Double widthConstraint, Double heightConstraint)
06-11 16:38:28.916 14158 14158 E AndroidRuntime:    at Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.IShapeView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Platform.MauiShapeView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].GetDesiredSize(Double widthConstraint, Double heightConstraint)
06-11 16:38:28.916 14158 14158 E AndroidRuntime:    at Microsoft.Maui.Layouts.LayoutExtensions.ComputeDesiredSize(IView view, Double widthConstraint, Double heightConstraint)
06-11 16:38:28.916 14158 14158 E AndroidRuntime:    at Microsoft.Maui.Controls.VisualElement.MeasureOverride(Double widthConstraint, Double heightConstraint)
06-11 16:38:28.916 14158 14158 E AndroidRuntime:    at Microsoft.Maui.Controls.VisualElement.Microsoft.Maui.IView.Measure(Double widthConstraint, Double heightConstraint)
06-11 16:38:28.916 14158 14158 E AndroidRuntime:    at Microsoft.Maui.Layouts.GridLayoutManager.GridStructure.EnsureFinalMeasure()
06-11 16:38:28.916 14158 14158 E AndroidRuntime:    at Microsoft.Maui.Layouts.GridLayoutManager.GridStructure.MeasureCells()
06-11 16:38:28.916 14158 14158 E AndroidRuntime:    at Microsoft.Maui.Layouts.GridLayoutManager.GridStructure..ctor(IGridLayout grid, Double widthConstraint, Double heightConstraint)
06-11 16:38:28.916 14158 14158 E AndroidRuntime:    at Microsoft.Maui.Layouts.GridLayoutManager.Measure(Double widthConstraint, Double heightConstraint)
06-11 16:38:28.916 14158 14158 E AndroidRuntime:    at Microsoft.Maui.Controls.Layout.CrossPlatformMeasure(Double widthConstraint, Double heightConstraint)
06-11 16:38:28.916 14158 14158 E AndroidRuntime:    at Microsoft.Maui.Platform.LayoutViewGroup.OnMeasure(Int32 widthMeasureSpec, Int32 heightMeasureSpec)
06-11 16:38:28.916 14158 14158 E AndroidRuntime:    at Android.Views.View.n_OnMeasure_II(IntPtr jnienv, IntPtr native__this, Int32 widthMeasureSpec, Int32 heightMeasureSpec)
06-11 16:38:28.916 14158 14158 E AndroidRuntime:    at Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PPII_V(_JniMarshal_PPII_V callback, IntPtr jnienv, IntPtr klazz, Int32 p0, Int32 p1)
```

Because `Easing.SpringIn` goes less than zero, we get the error "Width cannot be less than zero."

Can size less than zero be ignored? Is that what Xamarin.Forms did?

### Steps to Reproduce

Put a `BoxView` on screen and animated it with the code provided.

### Version with bug

6.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

11

### Did you find any workaround?

I can use Linear, but now the animation looks bad. Or I can change my math to never use a negative.

```
var inAnims = new Animation();
inAnims.Add(0, 1, new Animation(v => this.MilesProgressBar.WidthRequest = Math.Max(0,v), 0, 200, Easing.SpringIn));
inAnims.Add(0.25, 1, new Animation(v => this.SegmentsProgressBar.WidthRequest = Math.Max(0,v), 0, 100, Easing.SpringIn));

inAnims.Commit(this, "grow_bar", 16, 2000);
```

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Fix crash using animations and changing View sizes (negative values).

### Issues Fixed

Fixes #7953

### CODE REVIEW

**Comment by rmarinho** [comment]
Can we add a device test?

File: src/Controls/src/Core/HandlerImpl/VisualElement/VisualElement.Impl.cs [hunk 1]
```diff
@@ -180,12 +180,14 @@ Semantics IView.Semantics
 		internal Semantics SetupSemantics() =>
 			_semantics ??= new Semantics();
 
-		static void ValidatePositive(double value, string name)
+		static double ValidatePositive(double value)
```
Comment (hartez) [inline src/Controls/src/Core/HandlerImpl/VisualElement/VisualElement.Impl.cs:183]: This should probably be called `EnsurePositive` - it's not really a validation method anymore.

Comment (hartez) [inline src/Controls/src/Core/HandlerImpl/VisualElement/VisualElement.Impl.cs:183]: This should be renamed to something like `EnsurePositive` - it's not really a validation method anymore.

**Review by hartez — CHANGES_REQUESTED** [review]

File: src/Controls/src/Core/HandlerImpl/VisualElement/VisualElement.Impl.cs (hunk 1 above)
Comment (jsuarezruiz) [reply src/Controls/src/Core/HandlerImpl/VisualElement/VisualElement.Impl.cs]: Changed.

**Review by hartez — APPROVED** [review]

**Pull request merged**